solutions/instances/main.tf

```diff
@@ -79,6 +99,7 @@ locals {
   scc_cos_kms_key_crn = var.existing_scc_cos_bucket_name != null ? null : var.existing_scc_cos_kms_key_crn != null ? var.existing_scc_cos_kms_key_crn : module.kms[0].keys[format("%s.%s", local.scc_cos_key_ring_name, local.scc_cos_key_name)].crn
   cos_instance_crn    = var.existing_cos_instance_crn != null ? var.existing_cos_instance_crn : module.cos[0].cos_instance_crn
   cos_bucket_name     = var.existing_scc_cos_bucket_name != null ? var.existing_scc_cos_bucket_name : module.cos[0].buckets[local.scc_cos_bucket_name].bucket_name
+  cos_instance_guid   = var.existing_cos_instance_crn != null ? element(split(":", var.existing_cos_instance_crn), length(split(":", var.existing_cos_instance_crn)) - 3) : module.cos[0].cos_instance_guid
```

Comment:
we found issues when using `- 3` - do not use minus when parsing. Go forwards. There is now a CRN parser module at https://github.com/terraform-ibm-modules/terraform-ibm-common-utilities/tree/main/modules/crn-parser so can you use this instead?

Reply:
I think Steve found a bug with CRN parser that prevents it being used here currently, so for now, proceed with this approach, but don't work backwards